src/main/java/org/cactoos/collection/CollectionOf.java

```diff
@@ -60,13 +71,123 @@ public CollectionOf(final T... array) {
-            return list;
-        });
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param slr The scalar
+     */
+    public CollectionOf(final Scalar<Collection<T>> slr) {
+        this.col = new Unchecked<>(slr);
+    }
+
+    @Override
+    public int size() {
+        return this.col.value().size();
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return this.col.value().isEmpty();
+    }
+
+    @Override
+    public boolean contains(final Object obj) {
+        return this.col.value().contains(obj);
+    }
+
+    @Override
+    public Iterator<T> iterator() {
+        return this.col.value().iterator();
+    }
+
+    @Override
+    public Object[] toArray() {
+        return this.col.value().toArray();
+    }
+
+    @Override
+    public <X> X[] toArray(final X[] array) {
+        return this.col.value().toArray(array);
     }
 
+    @Override
+    public boolean add(final T elem) {
+        return this.col.value().add(elem);
+    }
+
+    @Override
+    public boolean remove(final Object obj) {
+        return this.col.value().remove(obj);
+    }
+
+    @Override
+    public boolean containsAll(final Collection<?> other) {
+        return this.col.value().containsAll(other);
+    }
+
+    @Override
+    public boolean addAll(final Collection<? extends T> other) {
+        return this.col.value().addAll(other);
+    }
+
+    @Override
+    public boolean removeAll(final Collection<?> other) {
+        return this.col.value().removeAll(other);
+    }
+
+    @Override
+    public boolean retainAll(final Collection<?> other) {
+        return this.col.value().retainAll(other);
+    }
+
+    @Override
+    public void clear() {
+        this.col.value().clear();
+    }
+
+    @Override
+    public String toString() {
+        return this.col.value().toString();
+    }
+
+    @Override
+    @SuppressFBWarnings("EQ_UNUSUAL")
+    public boolean equals(final Object other) {
+        return new Unchecked<>(
+            new Or(
+                () -> other == this,
+                new And(
+                    () -> other != null,
+                    () -> Collection.class.isAssignableFrom(other.getClass()),
+                    () -> {
+                        final Collection<?> compared = (Collection<?>) other;
+                        return this.size() == compared.size();
+                    },
+                    () -> {
+                        final Collection<?> compared = (Collection<?>) other;
+                        final Iterator<?> iterator = compared.iterator();
+                        return new Unchecked<>(
+                            new And(
+                                (T input) -> input.equals(iterator.next()),
+                                this
+                            )
+                        ).value();
+                    }
+                )
+            )
+        ).value();
+    }
+
+    @Override
+    public int hashCode() {
+        return new HashCode(this).value();
```

Comment:
@victornoel I don't think it the correct way to use `HashCode`. You must pass the attributes to the constructor, right? WDYT?

Reply:
@fabriciofx I've modified `HashCode` so that it can take an `Iterable` of attributes, so `this` is an `Iterable` of the elements of the collection.

Reply:
@victornoel sounds good to me